PaddleAudio/paddleaudio/models/wav2vec2/config_utils.py

```diff
@@ -0,0 +1,315 @@
+        """
+        From a ``pretrained_model_name_or_path``, resolve to a dictionary of parameters, to be used for instantiating a
+        :class:`PretrainedConfig` using ``from_dict``.
+
+        Parameters:
+            pretrained_model_name_or_path (:obj:`str` or :obj:`os.PathLike`):
+                The identifier of the pre-trained checkpoint from which we want the dictionary of parameters.
+
+        Returns:
+            :obj:`Tuple[Dict, Dict]`: The dictionary(ies) that will be used to instantiate the configuration object.
+
+        """
+
+        from_pipeline = kwargs.pop("_from_pipeline", None)
+        from_auto_class = kwargs.pop("_from_auto", False)
+
+        user_agent = {"file_type": "config", "from_auto_class": from_auto_class}
+        if from_pipeline is not None:
+            user_agent["using_pipeline"] = from_pipeline
+
+        local_files_only = True
+        pretrained_model_name_or_path = str(pretrained_model_name_or_path)
+        config_file = pretrained_model_name_or_path
+        # Load config dict
+        try:
+            config_dict = cls._dict_from_json_file(config_file)
+        except json.JSONDecodeError:
+            msg = (
+                f"Couldn't reach server at '{config_file}' to download configuration file or "
+                "configuration file is not a valid JSON file. "
+                f"Please check network or file content here: {config_file}.")
+            raise EnvironmentError(msg)
+
+        logger.info(f"loading configuration file {config_file}")
+
+        return config_dict, kwargs
+
+    @classmethod
+    def from_dict(cls, config_dict: Dict[str, Any],
+                  **kwargs) -> "PretrainedConfig":
+        """
+        Instantiates a :class:`PretrainedConfig` from a Python dictionary of parameters.
+
+        Args:
+            config_dict (:obj:`Dict[str, Any]`):
+                Dictionary that will be used to instantiate the configuration object. Such a dictionary can be
+                retrieved from a pretrained checkpoint by leveraging the
+                :func:`Wav2VePretrainedConfig.get_config_dict` method.
+            kwargs (:obj:`Dict[str, Any]`):
+                Additional parameters from which to initialize the configuration object.
+
+        Returns:
+            :class:`PretrainedConfig`: The configuration object instantiated from those parameters.
+        """
+        return_unused_kwargs = kwargs.pop("return_unused_kwargs", False)
+
+        config = cls(**config_dict)
+        to_remove = []
+        for key, value in kwargs.items():
+            if hasattr(config, key):
+                setattr(config, key, value)
+                to_remove.append(key)
+        for key in to_remove:
+            kwargs.pop(key, None)
+
+        #logger.info(f"Model config {config}")
+        if return_unused_kwargs:
+            return config, kwargs
+        else:
+            return config
+
+    @classmethod
+    def from_json_file(
```

Comment:
Do we need this function in real usage? Since we have already remove HF's base class. Is it possible to simply further?

Reply:
from_json_file is called by from_pretrained. The config file is in json format. I think this config is needed as we will do fine-tuning on mandarin soon.